vuln_checker/main.py

```diff
@@ -21,19 +25,99 @@ def set_to_list(obj):
     raise TypeError
 
 
-def main():
-    logging.basicConfig(level=logging.INFO)
+priorities = frozenset([
+    "CVE",
+    "GHSA"
+])
+
+VulnInfo = TypedDict('VulnInfo', {'aliases': set[str], 'sources': set[str]})
+VulnMap = dict[str, VulnInfo]
+
+
+def key_by_priority(keys: Iterable[str], default) -> tuple[str, set[str]]:
+    def find_identifier_by_prefix(k) -> str | None:
+        filtered = list(filter(lambda x: x.startswith(k), keys))
+        if len(filtered) > 0:
+            return filtered[0]
+        return None
+
+    key = list(map(find_identifier_by_prefix, priorities))[0]
+    if key is None:
+        key = default
+    aliases = set(keys) - {key}
+    return key, aliases
 
-    check_databases_folder(DEFAULT_DATABASES_PATH)
-    create_dir(VULNS_PATH)
+
+def create_map_of_vulnerabilities() -> VulnMap:
+    vulns: VulnMap = {}
 
     for database_name, model in load_vulnerabilities(DEFAULT_DATABASES_PATH):
-        database_path = os.path.join(VULNS_PATH, database_name)
-        create_dir(database_path)
+        database_path = VULNS_PATH / database_name
+        database_path.mkdir(exist_ok=True)
         file_name = os.path.join(database_path, model.id + '.json')
         with open(file_name, 'w') as f:
```

Comment:
if we already use `pathlib`, maybe should we use this one here too?
```suggestion
        database_path /= model.id + '.json'
        with database_path.open('w') as f:
```